tests/chainer_tests/training_tests/triggers_tests/test_once_trigger.py

```diff
@@ -16,74 +16,94 @@ def get_expected(num, call_on_first=True):
     return [i == 0 and call_on_first for i in six.moves.range(num)]
 
 
+def get_finished(num, call_on_first=True):
+    return [i > 0 or not call_on_first for i in six.moves.range(num)]
+
+
 @testing.parameterize(
     # basic
     {
         'iter_per_epoch': 5, 'call_on_resume': False, 'resume': 4,
         'expected': get_expected(7),
-        'expected_resume': get_expected(7, False)},
+        'expected_resume': get_expected(7, False),
+        'finished': get_finished(7),
+        'finished_resume': get_finished(7, False)},
```

Comment:
I now think the parameterization is redundant. Is it right that the essential parameters are only `iter_per_epoch`, `call_on_resume`, and `resume`? How about removing others from parameter lists and making them properties instead?

Reply:
Thank you.
Like this?